apps/base-docs/docs/building-with-base-and-foundry/introduction.md

```diff
@@ -0,0 +1,154 @@
+- *[chisel]*: Chisel is a solidity REPL. You can write solidity code directly.
+- *[anvil]*: A local Blockchain node for testing and development purposes.
+
+Using Foundry you can manage your dependencies, compile your project, run tests, deploy smart contracts and interact with the chain from the command-line and via Solidity scripts.
+
+For a deep dive on the Foundry features and full capabilities, check out the [Foundry Book].
+
+## Installing Foundry
+
+In order to install Foundry, you can use `Foundryup`, the Foundry's toolchain installer.
+
+To install `Foundryup` you have to run in the terminal:
+
+```bash
+$ curl -L https://foundry.paradigm.xyz | bash
+```
+
+After `Foundryup` is installed, you can now install `Foundry` by running:
+
+```bash
+$ foundryup
+```
+
+You can verify the installation by trying the following commands:
+
+```bash
+$ forge --version
+$ cast --version 
+$ chisel --version
+$ anvil --version
+```
+
+## My First Foundry Project
+
+To create a foundry project we can simply run:
+
+```bash
+$ forge init hello_foundry_in_base
+```
+
+This will create a foundry project with the following structure:
+
+```bash
+├── lib          # all the libraries installed will live in this folder
+├── script       # scripts folder, i.e deploy scripts
+├── src          # smart contracts folder
+├── test         # tests folder
+└── foundry.toml # foundry configuration file
+```
+
+You will also notice a `.gitsubmodules` file, this is because `Foundry` handles dependencies using [Git submodules].
+
+By default the Foundry structure will have the smart contracts in the `src` folder. This value can be changed using the `foundry.toml` configuration file.
+
+For instance:
+
+```bash
+[profile.default]
+src = 'contracts'
+```
+
+In order to compile the project, simply run:
+
+```bash
+forge build:
+```
+
+## Setting up Foundry with Base
+
+In order to work with `Base`, you need to configure a couple of settings in the configuration `foundry.toml` file.
+
+The first thing is the solidity version. 
+
+> As of December 2023, Base does not support yet the new PUSH0 opcode introduced in Shanghai, which is the default target for the Solidity compiler if you use version 0.8.20 or later. You can learn more about the [Differences between Ethereum and Base].
+
+We recommend using 0.8.19 or lower until this is upgraded. Keep in mind that the update has been deployed to Base Goerli and Base Sepolia.
+
+You need to configure your config file as follows:
+
+```bash
+[profile.default]
+src = 'src'
+out = 'out'
+libs = ['lib']
+solc_version = "0.8.19"
+```
+
+Be sure that you modify the pragma of your contracts and simply run `forge build` to ensure everything works well.
+
+Secondly, we recommend to setup the JSON RPC endpoints for base and the API key for [Basescan] in the configuration file, so once you decide to deploy your smart contracts, your environment will be fully ready.
```

Comment:
```suggestion
We also recommend setting up JSON RPC endpoints for Base and the API key for [Basescan] in the configuration file so that your environment is ready to deploy your smart contracts.
```